src/cmdstan/stansummary.cpp

```diff
@@ -147,7 +154,10 @@ Example:  stansummary model_chain_1.csv model_chain_2.csv
   }
   stan::io::stan_csv_metadata metadata = csv_parsed[0].metadata;
   std::vector<std::string> param_names = csv_parsed[0].header;
+
+  bool reorder_params = true;
   if (requested_params_vec.size() > 0) {
+    reorder_params = false;
```

Comment:
if a user has requested a series of parameters, then they should see them in the order that they were requested.
if the parameter is a container variable, the user needs to specify the elements that they want to see, so if they're asking for elements of a multi-dim container,  they are likely to specify them in row-major order anyhow.

Reply:
Not sure I follow, because at least experimentally the code seems to do something different. 

```shell
$ ./bin/stansummary -i y.1.1 -i y.2.2 -i y.2.1 -i y.1.3 ./src/test/interface/matrix_output.csv 
...
y[1,1]   0.43   0.12    0.39  0.15  0.059  0.16  0.93       5.0       5.0    1.4
y[1,3]   0.47   0.10    0.33  0.39  0.068  0.48  0.86       5.0       5.0   0.99
y[2,1]   0.51  0.087    0.27  0.38   0.17  0.49  0.90       5.0       5.0    1.4
y[2,2]   0.56  0.087    0.28  0.31   0.23  0.53  0.93       5.0       5.0   0.92
```

Based on your comment above, I would not expect them to be reordered. That said, having them re-ordered is fine, but I want to understand the codes intention

Reply:
Oh, it appears this is a indexing issue or perhaps a mis-labling, since y[2,2]'s average value is not actually 0.56, but 0.47...

Reply:
Hm, I think something is pretty wrong here. Here's the output on `develop`:

```
y[1,1]          0.43   0.12    0.39  0.023  0.17  0.93     10   5.7e+03    1.1
y[2,1]          0.51   -nan    0.27   0.17  0.49  0.93   -354  -2.0e+05   0.91
y[2,2]          0.56  0.087    0.28   0.16  0.65  0.99     10   5.7e+03   0.94
y[1,3]          0.47   -nan    0.33  0.045  0.71  0.94    -92  -5.2e+04   0.90
```

Reply:
So it looks like it _is_ reordering when passing `-i`, and something is wrong when not -- since you get different labels for the same rows

Reply:
redid the logic so that the order specified will be the order in which they are reported.